File: Happy_Olds_Web/src/MedicalBundle/Entity/Reponse.php
<?php

namespace MedicalBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use SBC\NotificationsBundle\Builder\NotificationBuilder;
use SBC\NotificationsBundle\Model\NotifiableInterface;

/**
 * Reponse
 *
 * @ORM\Table(name="reponse")
 * @ORM\Entity(repositoryClass="MedicalBundle\Repository\ReponseRepository")
 */
class Reponse implements NotifiableInterface
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @ORM\ManyToOne(targetEntity="Question", inversedBy="reponses")
     * @ORM\JoinColumn(name="question_id",referencedColumnName="id")
     */
    private $question;
    /**
     * @ORM\ManyToOne(targetEntity="\HappyOldsMainBundle\Entity\User")
     * @ORM\JoinColumn(name="user_id",referencedColumnName="id")
     */

    private $user;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateR", type="date")
     */


    private $dateR;

    /**
     * @var string
     *
     * @ORM\Column(name="text", type="string", length=255)
     */
    private $text;


    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set text
     *
     * @param string $text
     *
     * @return Reponse
     */
    public function setText($text)
    {
        $this->text = $text;

        return $this;
    }

    /**
     * Get text
     *
     * @return string
     */
    public function getText()
    {
        return $this->text;
    }

    /**
     * Set question
     *
     * @param \MedicalBundle\Entity\Question $question
     *
     * @return Reponse
     */
    public function setQuestion(\MedicalBundle\Entity\Question $question = null)
    {
        $this->question = $question;

        return $this;
    }

    /**
     * Get question
     *
     * @return \MedicalBundle\Entity\Question
     */
    public function getQuestion()
    {
        return $this->question;
    }

    /**
     * Set user
     *
     * @param \HappyOldsMainBundle\Entity\User $user
     *
     * @return Reponse
     */
    public function setUser(\HappyOldsMainBundle\Entity\User $user = null)
    {
        $this->user = $user;

        return $this;
    }

    /**
     * Get user
     *
     * @return \HappyOldsMainBundle\Entity\User
     */
    public function getUser()
    {
        return $this->user;
    }

    /**
     * Set dateR
     *
     * @param \DateTime $dateR
     *
     * @return Reponse
     */
    public function setDateR($dateR)
    {
        $this->dateR = $dateR;

        return $this;
    }

    /**
     * Get dateR
     *
     * @return \DateTime
     */
    public function getDateR()
    {
        return $this->dateR;
    }

    public function __construct()
    {
        $this->dateR = new \DateTime('now');
    }

    public function notificationsOnCreate(NotificationBuilder $builder)
    {
        // TODO: Implement notificationsOnCreate() method.
        $notification=New NotificationMedical();
        $notification->setTitle("Nouvelle reponse")
            ->setDescription($this->getText())
            ->setRoute('medical_detail')
            ->setParameters(array('id'=>$this->getId()));
        $builder->addNotification($notification);

        return $builder ;

    }
    public function notificationsOnUpdate(NotificationBuilder $builder)
    {
        // TODO: Implement notificationsOnUpdate() method.
        $notification=New NotificationMedical();
        $notification->setTitle("mise à jour reponse")
            ->setDescription($this->getText())
            ->setRoute('medical_detail')
            ->setParameters(array('id'=>$this->getId()));
        $builder->addNotification($notification);

        return $builder ;
    }
    public function notificationsOnDelete(NotificationBuilder $builder)
    {
        // TODO: Implement notificationsOnDelete() method.
        return $builder;
    }
}

File: Happy_Olds_mobile/src/tn/esprit/happyOlds/Medical/entity/Reponse.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.esprit.happyOlds.Medical.entity;

import java.util.Date;
import tn.esprit.happyOlds.entity.User;

/**
 *
 * @author Yousra Trabelsi
 */
public class Reponse {
    
    private int id ;
    private User user;
    private int question;
    private String text;
    private Date dateR;
    
    
     public Reponse() {
    }

    public Reponse(int id, User user, int question, String text, Date dateR) {
        this.id = id;
        this.user = user;
        this.question = question;
        this.text = text;
        this.dateR = dateR;
    }

    @Override
    public String toString() {
        return "Reponse{" + "id=" + id + ", user=" + user + ", question=" + question + ", text=" + text + ", dateR=" + dateR + '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public int getQuestion() {
        return question;
    }

    public void setQuestion(int question) {
        this.question = question;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getDateR() {
        return dateR;
    }

    public void setDateR(Date dateR) {
        this.dateR = dateR;
    }
     
    
}
